### ISSUE TITLE
can mask be supported?

### ISSUE THREAD

**Issue by unwritten** [ROCm/flash-attention#30]
in
flash_attn_func(q, k, v, dropout_p=0.0, softmax_scale=None, causal=False):

can mask be supported before going into softmax like:

            att = att.masked_fill(mask[:, None, :, :], min_value)
            att = F.softmax(att, dim=-1, dtype=torch.float32).to(rdtype).masked_fill(mask[:, None, :, :], 0.0)

**Comment by adityas-amd** [comment]
Hi @unwritten,
I think it is supported on main branch, https://github.com/ROCm/flash-attention/blob/99730ade130f8f8b00ae9bf912ec53578afdf5dd/flash_attn/flash_attn_interface.py#L77-L106
